salt/modules/nftables.py

```diff
@@ -330,7 +332,7 @@ def save(filename=None, family='ipv4'):
     try:
         with salt.utils.files.fopen(filename, 'w+') as _fh:
             # Write out any changes
-            _fh.writelines(rules)
+            _fh.writelines(salt.utils.data.decode(rules))
```

Comment:
This needs to be an `encode`, and we should be opening the file using a mode of `'wb'`.